src/thermo/src/4C_thermo_timint.cpp

```diff
@@ -379,11 +377,11 @@ void Thermo::TimInt::read_restart_state()
  *----------------------------------------------------------------------*/
 void Thermo::TimInt::write_runtime_output()
 {
-  runtime_output_writer_->reset();
+  runtime_output_writer_.value()->reset();
```

Comment:
```suggestion
  runtime_output_writer_->reset();
```
If you unconditionally access the value, just write this.

Also, I should add something like `FOUR_C_ASSERT(runtime_output_writer_.has_value(), "Runtime output not initialized.")`. If the function is public, it should probably even be `FOUR_C_ASSERT_ALWAYS`.